### PR TITLE
Wind feed Oracle

### PR DESCRIPTION
**Pull request by janus**
This is my Wind Feed Oracle. 
It is live on kovan and completed. 
https://explorer.iex.ec/kovan/deal/0x7c9612d23545e8bef8fa76b9deb64398537c9c7c0ee0c8a8f38b15f7bbecdd42

### CODE REVIEW

**Comment by Amxx** [comment]
Results for the run on kovan is a file, this does not match the expected result of an oracle (which results must be encoded so that they are processed onchain)

https://explorer.iex.ec/kovan/task/0xf0881ff55a30eadd44c3dc081b21692b36a963aff457b3be5bc7913bc1da4ac1

File: windy-feed/app/wind-feed.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+ *****************************************************************************/
+
+
+var lat = lat || 37.39;
+var long = long || -122.08;
+
+
+const url = "https://api.openweathermap.org/data/2.5/weather";
+const query1 = `${url}?lat=${lat}&lon=${long}&appid=${APIKEY1}`;
+
```
Comment (Amxx) [inline windy-feed/app/wind-feed.js]: This is good but not deterministic. I would suggest replacing lat/lon with lat/lon/time and use the endpoint:
`http://history.openweathermap.org/data/2.5/history/city?lat={lat}&lon={lon}&type=hour&start={start}&cnt={cnt}`

File: windy-feed/app/wind-feed.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    // subtract local time zone offset
+    // get UTC time in msec
+    var utc = d.getTime() + (d.getTimezoneOffset() * 60000);
+
+    // create new Date object for different city
+    // using supplied offset
+    var nd = new Date(utc + (1000 * results.timezone));
+
+
+	timestamp = nd.getTime();
```
Comment (Amxx) [inline windy-feed/app/wind-feed.js]: If moving from lat/lon to lat/lon/time → use the time as a timestamp (deterministic)

File: windy-feed/app/wind-feed.js (hunk 1 above)
Comment (janus) [reply windy-feed/app/wind-feed.js]: @Amxx 
Thanks for your review. But historical data is not for free subscription . https://openweathermap.org/faq#error401
I don't know what to do next.

**Comment by sulliwane** [comment]
Ok @janus, that will work with live data then! great work by the way. Expect the bounties to be released in two weeks maximum (hopefully before that), thanks!

**Comment by janus** [comment]
@sulliwane 
Let me know when the time comes so that I would re-apply. I have limited gitcoin engagement rights .. because of the time this bounty has taken.. I removed myself to use it on other bounties

**Pull request merged**